Tests/ServiceDiscoveryTests/TypeErasedServiceDiscoveryTests.swift

```diff
@@ -134,42 +122,50 @@ class TypeErasedServiceDiscoveryTests: XCTestCase {
         let anyServiceDiscovery = AnyServiceDiscovery(serviceDiscovery)
 
         let semaphore = DispatchSemaphore(value: 0)
-        let resultCounter = ManagedAtomic<Int>(0)
-
-        // Two results are expected:
-        // Result #1: LookupError.unknownService because bar-service is not registered
-        // Result #2: Later we register bar-service and that should notify the subscriber
-        anyServiceDiscovery.subscribe(
-            to: self.barService,
-            onNext: { result in
-                resultCounter.wrappingIncrement(ordering: .relaxed)
-
-                guard resultCounter.load(ordering: .relaxed) <= 2 else {
-                    return XCTFail("Expected to receive result 2 times only")
-                }
+        let counter = ManagedAtomic<Int>(0)
+
+        Task.detached {
+            // Allow time for subscription to start
+            usleep(100_000)
+            // Update #1
+            serviceDiscovery.register(self.barService, instances: [])
+            usleep(50000)
```

Comment:
Technically we have `await Task.sleep(nanoseconds:)` if you prefer that, but either is fine I suppose, its just a test